### PR TITLE
Add sorting for bank transactions

### PR DESCRIPTION
**Pull request by dphilipov**
Related to [#1588](https://github.com/podkrepi-bg/frontend/issues/1588)

## Motivation and context

* My initial solution was to sort the data on the front-end, but since the data is paginated, this sorts only the current page. This PR makes sure the sort details reach the query, thus sorting all bank transactions.

### CODE REVIEW

File: apps/api/src/bank-transactions/bank-transactions.service.ts [hunk 1]
```diff
@@ -65,6 +67,11 @@ export class BankTransactionsService {
       },
       skip: pageIndex && pageSize ? pageIndex * pageSize : undefined,
       take: pageSize ? pageSize : undefined,
+      orderBy: [
+        {
+          [sortBy as string]: sortOrder,
+        },
+      ],
```
Comment (igoychev) [inline apps/api/src/bank-transactions/bank-transactions.service.ts:74]: might be good to set a default sorting if the param is not sent by frontend, something like:
orderBy: [sortBy ? { [sortBy]: sortOrder ? sortOrder : 'desc' } : { transactionDate: 'desc' }],

**Review by igoychev — COMMENTED** [review]
looks great!
just please check if the frontend is always passing the sortBy and sortOrder or add some defaults like in the comment below

File: apps/api/src/bank-transactions/bank-transactions.service.ts (hunk 1 above)
Comment (dphilipov) [reply apps/api/src/bank-transactions/bank-transactions.service.ts]: Implemented :+1:

**Review by igoychev — APPROVED** [review]
wonderful!

**Pull request merged**